### ISSUE TITLE
Remove IKestrelTrace

### ISSUE THREAD

**Issue by JamesNK** [dotnet/aspnetcore#31581]
[IKestralTrace](https://github.com/dotnet/aspnetcore/blob/7b04ecf2ccc15b99da2cca886960b44a7e640e3d/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs#L14) was added before it was easily possible to unit test logs.

See https://github.com/dotnet/aspnetcore/pull/31562#issuecomment-814296577

We should remove the interface and change [KestrelTrace](https://github.com/dotnet/aspnetcore/blob/main/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs) into a static class. Will save making interface method calls.

Will likely require a lot of changes to unit tests.

Repeat for ISocketTrace and IQuicTrace.

**Comment by JamesNK** [comment]
@davidfowl @halter73

**Comment by davidfowl** [comment]
@gfoidl want to take a stab at this?

**Comment by gfoidl** [comment]
I'll give it a try 😃

**Comment by gfoidl** [comment]
Once https://github.com/dotnet/aspnetcore/pull/31596 is merged, I'm starting, otherwise there'll be a merge conflict.

**Comment by fvoronin** [comment]
@gfoidl, JFYI, #31596 was merged

**Comment by gfoidl** [comment]
Thanks for the info!
Will start soon working on it -- and will look into https://github.com/dotnet/aspnetcore/issues/32087 as well.

**Comment by gfoidl** [comment]
Update: I'd like to wait on https://github.com/dotnet/runtime/issues/51927. This unblocks the usage here for source generators.

**Comment by adityamandaleeka** [comment]
@gfoidl Are you still interested in taking on this issue?

**Comment by JamesNK** [comment]
Also, Kestrel and transport logs have been updated to use source generators.

https://github.com/dotnet/aspnetcore/pull/34910

**Comment by gfoidl** [comment]
Yes, my interest is still there, but I'm a bit sick at the moment (and my brain not working at the level I'd like it to work). So I won't start before next week.

**Comment by adityamandaleeka** [comment]
@gfoidl Sorry to hear that you're not feeling well, and hope you get better soon. Please don't feel rushed to work on this until you are fully recovered.

I'm going to change the milestone for now just for tracking purposes. We can add a milestone again once we know when it'll land.
